engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```diff
@@ -2046,6 +2207,51 @@ private void verifyLiveMigrationMapForKVM(Map<VolumeInfo, DataStore> volumeDataS
             if (destStoragePoolVO == null) {
                 throw new CloudRuntimeException("Destination storage pool with ID " + dataStore.getId() + " was not located.");
             }
+
+            if (storageTypeConsistency == null) {
+                storageTypeConsistency = destStoragePoolVO.isManaged();
+            } else if (storageTypeConsistency != destStoragePoolVO.isManaged()) {
+                throw new CloudRuntimeException("Destination storage pools must be either all managed or all not managed");
+            }
+
+            if (!destStoragePoolVO.isManaged()) {
+                if (srcStoragePoolVO.getPoolType() != Storage.StoragePoolType.NetworkFilesystem ||
+                        destStoragePoolVO.getPoolType() != Storage.StoragePoolType.NetworkFilesystem) {
```

Comment:
This verification blocks the migration between local storage (storage type filesystem), allowed with PR #2997. 
It should allow migrating also to Storage.StoragePoolType.Filesystem.